### DISCUSSION TITLE
Show number of files and folders in overview and interactive mode

### DISCUSSION THREAD

**Discussion by LostInCompilation** [Byron/dua-cli#252] [Ideas]
It would be really amazing, if **dua** could show the number of files and folders per displayed directory next to the size. Also in interactive mode this would be greatly appreciated.

Thanks :)

**Comment by Byron** [comment]
Thanks for the suggestion. What helps with feature requests is to elaborate on the workflow that it would support.

In interactive mode, this already works today to some extend. Press `C` to add the 'count' column, displaying the amount of entries, i.e. directories + files + symlinks . Press `?` for all the hotkeys.

<img width="587" alt="Screenshot 2024-07-26 at 07 17 04" src="https://github.com/user-attachments/assets/51472099-7c0b-42bb-b8fe-a6ea49dd7e24">

What this issue is asking it to split the entry count into files & folders.

Further, it asks to add such counts in the non-interactive version.

<img width="229" alt="Screenshot 2024-07-26 at 07 19 10" src="https://github.com/user-attachments/assets/8df16004-d7cd-4816-93d4-e4cd509c5596">

**Comment by LostInCompilation** [comment]
Thanks for the info. I think it would be convenient to show the total number of files and folders directly in interactive mode by default, with the option to hide them when pressing c. In most terminals there is enough horizontal space available.

Splitting the count into files and folders would be good too.

**Reply by Byron** [reply]
I moved this issue into discussions as I find it too unfocussed now to act on it. The idea is to use this discussion to figure out what makes sense to create as an issue.

> Thanks for the info. I think it would be convenient to show the total number of files and folders directly in interactive mode by default, with the option to hide them when pressing c. In most terminals there is enough horizontal space available.

With that said, I'd like to keep the interactive focussed on the size, but also think that the initial state should be configurable. I wouldn't go all in with configuration files and such, but would want to start by providing an argument to `dua i -?` where `-?` is able to control which columns are shown (or which additional columns are shown maybe. Then people can create an alias.

This might even translate to `dua a`, the text-only form to allow showing counts there as well.

If this seems feasible, I welcome you to open an issue to describe the feature, maybe someone will come along an implement it.
